src/main/java/com/iexec/commons/poco/chain/IexecHubAbstractService.java

```diff
@@ -918,7 +915,9 @@ Optional<TaskDescription> repeatGetTaskDescriptionFromChain(String chainTaskId,
     }
 
     public boolean isTeeTask(String chainTaskId) {
-        final TaskDescription taskDescription = getTaskDescriptionFromChain(chainTaskId).orElse(null);
+        // Magical non-null retry delay to ease testing and because there are no retries
+        final TaskDescription taskDescription = repeatGetTaskDescriptionFromChain(chainTaskId, 1000, 0)
```

Comment:
Shouldn't we use `getTaskDescription` to avoid calling the chain if we already have the task description in cache?

Reply:
I agree with your point. Before taking a decision, I think some cleanup is needed on the application side.
The cache mechanism is not deployed properly in all applications and all applications do not receive proper cleanup signals. I preferred to avoid modifying the algorithm and allow ourselves some time to properly think the whole system.